catalog/rest/catalog-rest-impl/src/test/java/org/codice/ddf/rest/impl/CatalogServiceImplTest.java

```diff
@@ -656,120 +638,82 @@ BundleContext getBundleContext() {
     assertThat(attachmentInfoAndMetacard.getRight().getMetadata(), equalTo("Some Text Again"));
     assertThat(attachmentInfoAndMetacard.getRight().getAttribute("foo"), equalTo(null));
   }
-  /**
-   * Tests local retrieve with a null QueryResponse
-   *
-   * @throws Exception
-   */
-  @Test
-  public void testGetDocumentLocalNullQueryResponse() throws Exception {
-
-    CatalogFramework framework = givenCatalogFramework(SAMPLE_ID);
-    String transformer = mockTestSetup(framework, TestType.QUERY_RESPONSE_TEST);
-    Response response = executeTest(framework, transformer, true, null);
-    assertEquals(response.getStatus(), NOT_FOUND);
-    assertEquals(response.getEntity(), "<pre>Unable to retrieve requested metacard.</pre>");
-  }
 
-  /**
-   * Tests federated retrieve with a null QueryResponse
-   *
-   * @throws Exception
-   */
   @Test
-  public void testGetDocumentFedNullQueryResponse() throws Exception {
-
-    CatalogFramework framework = givenCatalogFramework(SAMPLE_ID);
-    String transformer = mockTestSetup(framework, TestType.QUERY_RESPONSE_TEST);
-    Response response = executeTest(framework, transformer, false, null);
-    assertEquals(response.getStatus(), NOT_FOUND);
-    assertEquals(response.getEntity(), "<pre>Unable to retrieve requested metacard.</pre>");
-  }
-
-  /**
-   * Tests local retrieve with a null Metacard
-   *
-   * @throws Exception
-   */
-  @Test
-  public void testGetDocumentLocalNullMetacard() throws Exception {
-
-    CatalogFramework framework = givenCatalogFramework(SAMPLE_ID);
-    String transformer = mockTestSetup(framework, TestType.METACARD_TEST);
-    Response response = executeTest(framework, transformer, true, null);
-    assertEquals(response.getStatus(), NOT_FOUND);
-    assertEquals(response.getEntity(), "<pre>Unable to retrieve requested metacard.</pre>");
-  }
-
-  /**
-   * Tests federated retrieve with a null Metacard
-   *
-   * @throws Exception
-   */
-  @Test
-  public void testGetDocumentFedNullMetacard() throws Exception {
+  @SuppressWarnings({"unchecked"})
+  public void testParsePartsTooLarge()
```

Comment:
❓ Should we also have a `testParsePartsWithAttributeOverrides` test to pair with `testParseAttachmentsWithAttributeOverrides`?